pkg/manager/member/tiproxy_member_manager.go

```diff
@@ -607,6 +537,209 @@ func (m *tiproxyMemberManager) getNewStatefulSet(tc *v1alpha1.TidbCluster, cm *c
+		volMounts = append(volMounts, corev1.VolumeMount{
+			Name: "tiproxy-tls", ReadOnly: true, MountPath: tiproxyHTTPServerPath,
+		})
+
+		vols = append(vols, corev1.Volume{
+			Name: "tiproxy-tls", VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: util.ClusterTLSSecretName(tc.Name, label.TiProxyLabelVal),
+				},
+			},
+		})
+	}
+	return vols, volMounts
+}
+
+func (m *tiproxyMemberManager) buildVolumesAndVolumeMountsForTLSV1(tc *v1alpha1.TidbCluster) ([]corev1.Volume, []corev1.VolumeMount) {
+	var (
+		vols      []corev1.Volume
+		volMounts []corev1.VolumeMount
+	)
+
+	tlsCluster := tc.IsTLSClusterEnabled()
+	tlsTiDB := tc.Spec.TiDB != nil && tc.Spec.TiDB.IsTLSClientEnabled()
+	if tlsCluster {
+		// This cert is not directly used by tiproxy itself, but mount it is useful for debug purpose.
+		// the common client mTLS cert
+		volMounts = append(volMounts, corev1.VolumeMount{
+			Name:      util.ClusterClientVolName,
+			ReadOnly:  true,
+			MountPath: util.ClusterClientTLSPath,
+		})
+		vols = append(vols, corev1.Volume{
+			Name: util.ClusterClientVolName, VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: util.ClusterClientTLSSecretName(tc.Name),
+				},
+			},
+		})
+
+		// inter-components (cluster) mTLS cert for tiproxy
+		volMounts = append(volMounts, corev1.VolumeMount{
+			Name:      tiProxyClusterCertVolumeMount,
+			ReadOnly:  true,
+			MountPath: tiproxyClusterCertPath,
+		})
+		vols = append(vols, corev1.Volume{
+			Name: tiProxyClusterCertVolumeMount, VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: util.ClusterTLSSecretName(tc.Name, label.TiProxyLabelVal),
+				},
+			},
+		})
+	}
+
+	// use the same server-side cert at SQL port (default 4000) for tiproxy and tidb
+	if tlsTiDB {
+		volMounts = append(volMounts, corev1.VolumeMount{
+			Name: "tidb-server-tls", ReadOnly: true, MountPath: tiproxyServerPath,
+		})
+
+		vols = append(vols, corev1.Volume{
+			Name: "tidb-server-tls", VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: util.TiDBServerTLSSecretName(tc.Name),
+				},
+			},
+		})
+	}
+
+	return vols, volMounts
+}
+
+func (m *tiproxyMemberManager) modifyConfigMapForTLSV1(tc *v1alpha1.TidbCluster, cfgWrapper *v1alpha1.TiProxyConfigWraper) {
+	tlsCluster := tc.IsTLSClusterEnabled()
+	tlsTiDB := tc.Spec.TiDB != nil && tc.Spec.TiDB.IsTLSClientEnabled()
+	if tlsCluster {
+		cfgWrapper.Set("security.cluster-tls.ca", path.Join(tiproxyClusterCertPath, "ca.crt"))
+		cfgWrapper.Set("security.cluster-tls.key", path.Join(tiproxyClusterCertPath, "tls.key"))
+		cfgWrapper.Set("security.cluster-tls.cert", path.Join(tiproxyClusterCertPath, "tls.crt"))
+	}
+
+	if tlsTiDB {
+		cfgWrapper.Set("security.server-tls.key", path.Join(tiproxyServerPath, "tls.key"))
+		cfgWrapper.Set("security.server-tls.cert", path.Join(tiproxyServerPath, "tls.crt"))
+		// fixme(chenfei): why?
```

Comment:
@xhebox PTAL   Can we remove it?

Reply:
`If skip-ca is true with non-empty ca, server will only verify clients if it can provide any cert. Otherwise, clients must provide a cert.`

That is for server tls, configure `.ca` will require `mTLS`. But if you also set `skip-ca = true`, it becomes optional.

Check https://github.com/pingcap/tiproxy/blob/d8c7aab110f15c2eee3c077e2e7c72e2cbabc940/conf/proxy.toml#L72-L89C68.